src/course-outline/configure-modal/ConfigureModal.jsx

```diff
@@ -8,81 +8,295 @@ import {
+      if (selectedPartitionIndex >= 0) {
+        const partitionId = userPartitionInfo.selectablePartitions[selectedPartitionIndex].id;
+        groupAccess[partitionId] = selectedGroups.map(g => parseInt(g, 10));
+      }
+      onConfigureSubmit(isVisibleToStaffOnly, groupAccess);
+      break;
+    default:
+      break;
+    }
   };
 
-  return (
-    <ModalDialog
-      className="configure-modal"
-      isOpen={isOpen}
-      onClose={onClose}
-      hasCloseButton
-      isFullscreenOnMobile
-    >
-      <ModalDialog.Header className="configure-modal__header">
-        <ModalDialog.Title>
-          {intl.formatMessage(messages.title, { title: displayName })}
-        </ModalDialog.Title>
-      </ModalDialog.Header>
-      <ModalDialog.Body className="configure-modal__body">
+  const renderModalBody = () => {
+    switch (category) {
+    case COURSE_BLOCK_NAMES.chapter.id:
+      return (
+        <Tabs>
+          <Tab eventKey="basic" title={intl.formatMessage(messages.basicTabTitle)}>
+            <BasicTab
+              releaseDate={releaseDate}
+              setReleaseDate={setReleaseDate}
+              isSubsection={isSubsection}
+              graderType={graderType}
+              courseGraders={courseGraders === 'undefined' ? [] : courseGraders}
+              setGraderType={setGraderType}
+              dueDate={dueDateState}
+              setDueDate={setDueDateState}
+            />
+          </Tab>
+          <Tab eventKey="visibility" title={intl.formatMessage(messages.visibilityTabTitle)}>
+            <VisibilityTab
+              category={category}
+              isSubsection={isSubsection}
+              isVisibleToStaffOnly={isVisibleToStaffOnly}
+              setIsVisibleToStaffOnly={setIsVisibleToStaffOnly}
+              showWarning={visibilityState === VisibilityTypes.STAFF_ONLY}
+              hideAfterDue={hideAfterDueState}
+              setHideAfterDue={setHideAfterDueState}
+              showCorrectness={showCorrectnessState}
+              setShowCorrectness={setShowCorrectnessState}
+            />
+          </Tab>
+        </Tabs>
+      );
+    case COURSE_BLOCK_NAMES.sequential.id:
+      return (
         <Tabs>
           <Tab eventKey="basic" title={intl.formatMessage(messages.basicTabTitle)}>
-            <BasicTab releaseDate={releaseDate} setReleaseDate={setReleaseDate} />
+            <BasicTab
+              releaseDate={releaseDate}
+              setReleaseDate={setReleaseDate}
+              isSubsection={isSubsection}
+              graderType={graderType}
+              courseGraders={courseGraders === 'undefined' ? [] : courseGraders}
+              setGraderType={setGraderType}
+              dueDate={dueDateState}
+              setDueDate={setDueDateState}
+            />
           </Tab>
           <Tab eventKey="visibility" title={intl.formatMessage(messages.visibilityTabTitle)}>
```

Comment:
When viewing this tab I would expect the modal to scroll instead of the entire page. The expected scroll behavior can be seen in the highlights modal